packages/frontend/src/data/bugsnag/index.ts

```diff
@@ -1,14 +1,21 @@
 import bugsnag from '@bugsnag/expo';
 import { BUGSNAG_API_KEY } from 'react-native-dotenv';
+import * as BugsnagCore from '@bugsnag/expo/dist/types/bugsnag-core';
+import { ErrorReporterParams } from '../../domain/ErrorReporter';
 
-export const bugsnagClient = bugsnag(BUGSNAG_API_KEY);
-
-type BugSnagInit = {
+export class BugSnag implements ErrorReporterParams<BugsnagCore.Client> {
```

Comment:
Ora ora ora, sintax-feia-aparte, a ideia aqui eh fazer uso de algo chamado [Dependency inversion](https://en.wikipedia.org/wiki/Dependency_inversion_principle).

Dizer que a mae dependende financeiramente do filho NAO eh o mesmo que dizer que o filho depende financeiramente da mae.

Neste caso, dizer que a camada de dominio depende da camada de dados nao eh o mesmo que dizer que a camada de dados depende da camada de dominio. Por isso o nome inversao de dependencia, ao invez de um depender do outro tu mexe co codigo para o outro depender do um.

Mas o que isso ajuda no fim das contas?
Quanto mais puro o seu dominio for, mais facil de ajustar sem causar merdas sera.

Isso vai parecer confuso de inicio, mas reflete um pouco sobre isso.